pkg/bootloader/grub.go

```diff
@@ -0,0 +1,233 @@
+		for _, name := range []string{"shim.efi", "grub.efi", "MokManager.efi"} {
+			src := filepath.Join(srcDir, name)
+			target := filepath.Join(targetDir, name)
+			err = vfs.CopyFile(g.s.FS(), src, target)
+			if err != nil {
+				g.s.Logger().Error("Error copying EFI app '%s': %s", src, err.Error())
+				return err
+			}
+		}
+
+		// Copy grub config if it exists.
+		grubCfg := filepath.Join(rootPath, "usr/lib/elemental/bootloader/grub.cfg")
+		err = vfs.CopyFile(g.s.FS(), grubCfg, targetDir)
+		if err != nil {
+			g.s.Logger().Warn("Failed copying grub.cfg: %s, skipping", err.Error())
+		}
+	}
+
+	return nil
+}
+
+// installGrub installs grub themes and configs to $ESP/grub2
+func (g *Grub) installGrub(rootPath string, esp *deployment.Partition) error {
+	g.s.Logger().Info("Syncing grub2 directory to ESP...")
+
+	// Since we are copying to a vfat filesystem we have to skip symlinks.
+	r := rsync.NewRsync(g.s, rsync.WithFlags("--archive", "--recursive", "--no-links"))
+
+	err := r.SyncData(filepath.Join(rootPath, "/usr/share/grub2"), filepath.Join(rootPath, esp.MountPoint, "grub2"))
+	if err != nil {
+		g.s.Logger().Error("Error syncing grub files: %s", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+// installKernelInitrd copies the kernel to the ESP and generates an initrd using dracut.
+func (g *Grub) installKernelInitrd(rootPath string, esp *deployment.Partition) (grubBootEntry, error) {
+	g.s.Logger().Info("Installing kernel/initrd")
+
+	osVars, err := vfs.LoadEnvFile(g.s.FS(), filepath.Join(rootPath, OsReleasePath))
+	if err != nil {
+		g.s.Logger().Info("Error loading %s vars: %s", OsReleasePath, err.Error())
+		return grubBootEntry{}, err
+	}
+
+	var (
+		osID string
+		ok   bool
+	)
+	if osID, ok = osVars["ID"]; !ok {
+		g.s.Logger().Error("Error /etc/os-release ID var not set.")
+		return grubBootEntry{}, fmt.Errorf("/etc/os-release ID not set")
+	}
+
+	kernel, kernelVersion, err := vfs.FindKernel(g.s.FS(), rootPath)
+	if err != nil {
+		g.s.Logger().Info("Error loading finding kernel: %s", err.Error())
+		return grubBootEntry{}, err
+	}
+
+	targetDir := filepath.Join(rootPath, esp.MountPoint, osID, kernelVersion)
+	err = vfs.MkdirAll(g.s.FS(), targetDir, vfs.DirPerm)
+	if err != nil {
+		g.s.Logger().Info("Error creating kernel dir '%s': %s", targetDir, err.Error())
+		return grubBootEntry{}, err
+	}
+
+	err = vfs.CopyFile(g.s.FS(), kernel, targetDir)
+	if err != nil {
+		g.s.Logger().Info("Error copying kernel '%s': %s", targetDir, err.Error())
+		return grubBootEntry{}, err
+	}
+
+	expectedInitrdPath := filepath.Join(filepath.Dir(kernel), Initrd)
+	if exists, _ := vfs.Exists(g.s.FS(), expectedInitrdPath); !exists {
+		return grubBootEntry{}, fmt.Errorf("initrd not found")
+	}
+
+	err = vfs.CopyFile(g.s.FS(), expectedInitrdPath, targetDir)
+	if err != nil {
+		g.s.Logger().Info("Error copying initrd '%s': %s", targetDir, err.Error())
+		return grubBootEntry{}, err
+	}
+
+	displayName, ok := osVars["PRETTY_NAME"]
```

Comment:
starting with SLES16 we actually have the displayName in `VARIANT` instead of PRETTY_NAME. can we handle that here?

Reply:
Thanks for the info! I added a check so the order goes PRETTY_NAME > VARIANT > NAME.